lib/libspl/include/assert.h

```diff
@@ -42,6 +42,17 @@ __assert_c99(const char *expr, const char *file, int line, const char *func)
 }
 #endif  /* __assert_c99 */
 
+#ifndef __assert
```

Comment:
How about something a little more drastic.  This has always been something of a mess we mostly inherited and there's not much reason to keep it as is.  Let's throw it out and do something a little more like the implementation in spl.ko where we avoid depending on any additional functionality beyond `panic()`, but in this case use `abort()`.  That should minimize the surface area for portability issues and standardize the ASSERT/VERIFY messages between user and kernel space.

Would something like this work for Musl?  behlendorf/zfs@86d328088b2437420646bd7f90a16f2b444cd0ee Note I only compile tested this so it may still need some work.